File: server/src/routes/service.route.ts
import express from 'express';
import {
    createService,
    deleteService,
    getAllServices,
    getServiceById,
    updateService,
} from '../controller/service.controller';
import { validate } from '../middleware';
import { CreateServiceType } from '../types';

const router = express.Router();

router
    .route('/')
    .get(getAllServices)
    .post(validate(CreateServiceType), createService);

router
    .route('/:id')
    .get(getServiceById)
    .put(validate(CreateServiceType), updateService)
    .delete(deleteService);

export default router;

File: server/routes/serviceRoute.go
package routes

import (
	"care-taker/controllers"
	"care-taker/middlewares"

	"github.com/gin-gonic/gin"
)

func ServiceRoutes(r *gin.RouterGroup) {
	r.GET("/", controllers.GetAllServices)
	r.POST("/", controllers.CreateService)
	r.GET("/:id", controllers.GetServiceById)
	r.PUT("/:id", middlewares.IsAdmin, controllers.UpdateService)
	r.DELETE("/:id", middlewares.IsAdmin, controllers.DeleteService)
}
